### PR TITLE
first pass at claro.vim

### PR DESCRIPTION
**Pull request by jzwood**
I originally added in all the keywords but it was too visually noisy for me. having specialized colors for the most used keywords seemed like a reasonable first step.

Screenshot 2023-05-18 at 10 23 24 PM

### CODE REVIEW

File: claro-vim/syntax/claro.vim [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+syntax match claroOperator "\v\<-"
+syntax match claroOperator "\v\>"
+syntax match claroOperator "\v\<"
+syntax match claroOperator "\v\>\="
+syntax match claroOperator "\v\<\="
+syntax match claroOperator "\v\=\="
+
+syntax match claroIdentifier "\v[a-zA-Z0-9]*"
+
+syntax match claroFunction "\v^[a-z][a-zA-Z0-9]*"
```
Comment (JasonSteving99) [inline claro-vim/syntax/claro.vim:45]: Claro also supports underscores in identifiers, here's the [regex the lexer uses](https://github.com/JasonSteving99/claro-lang/blob/4987344526b1582ce3d202eeb1ac0a3c98e97791/src/java/com/claro/ClaroLexer.flex#L88). Would you mind adding that?

**Review by JasonSteving99 — CHANGES_REQUESTED** [review]
This is awesome, thank you for contributing this! I got it working locally and makes a world of difference over plain white text I would've seen before if using vim.

I had one comment to support underscores in identifiers but otherwise looks great.

File: claro-vim/syntax/claro.vim (hunk 1 above)
Comment (jzwood) [reply claro-vim/syntax/claro.vim]: sure thing!

File: claro-vim/syntax/claro.vim [hunk 2]
```diff
@@ -40,23 +40,20 @@ syntax match claroOperator "\v\>\="
...
 
-syntax match claroIdentifier "\v[a-zA-Z0-9]*"
-
-syntax match claroFunction "\v^[a-z][a-zA-Z0-9]*"
+syntax match claroIdentifier "\v^[a-zA-Z][_a-zA-Z0-9]*"
 
 syntax region claroComment start="#" end="\n"
 
 highlight link claroIdentifier Identifier
+highlight link claroFunction Identifier
```
[the remark this answers is not in the document]
Comment (JasonSteving99) [reply claro-vim/syntax/claro.vim:48]: Ya you're right, there's just the one pattern. Do you need to drop the highlight link at line 48 now that you dropped the separate pattern match for claroFunction?

[the remark this answers is not in the document]
Comment (jzwood) [reply claro-vim/syntax/claro.vim]: removed!

**Review by JasonSteving99 — APPROVED** [review]

**Pull request merged**